src/ast/semantic_analyser.cpp

```diff
@@ -715,6 +715,19 @@ void SemanticAnalyser::visit(Call &call)
 #else
     error("BPF_FUNC_send_signal not available for your kernel version",
           call.loc);
+#endif
+  }
+  else if (call.func == "fdpath")
+  {
+#ifdef HAVE_GET_FD_PATH
+    if (check_varargs(call, 1, 1))
+    {
+      check_arg(call, Type::integer, 0);
```

Comment:
Does fdpath work for all probe types or should we restrict it? E.g `i:s:1 { @[pid,fdpath(1)]++ }`

Reply:
I added the check.. it's supported by tracepoints, I'm using kernel patch that adds it for kprobes,which I'll post

however the helper still did not get merged.. moreover got outrun by another one, so the ID changed ;-)) I'll wait with new push until it's merged, to avoid any confusion